authz/rbac_translator_test.go

```diff
@@ -298,6 +651,90 @@ func TestTranslatePolicy(t *testing.T) {
 			}`,
 			wantErr: `"allow_rules" 0: "headers" 0: unsupported "key" :method`,
 		},
+		"bad audit condition": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [
+				{
+					"name": "allow_authenticated",
+					"source": {
+						"principals":["*", ""]
+					}
+				}],
+				"audit_logging_options": {
+					"audit_condition": "ABC",
+					"audit_loggers": [
+						{
+							"name": "stdout_logger",
+							"config": {},
+							"is_optional": false
+						}
+					]
+				}
+			}`,
+			wantErr: `Failed to parse AuditCondition ABC`,
+		},
+		"bad audit logger config": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [
+				{
+					"name": "allow_authenticated",
+					"source": {
+						"principals":["*", ""]
+					}
+				}],
+				"audit_logging_options": {
+					"audit_condition": "NONE",
+					"audit_loggers": [
+						{
+							"name": "stdout_logger",
+							"config": "abc",
+							"is_optional": false
+						}
+					]
+				}
+			}`,
+			wantErr: `failed to unmarshal policy`,
```

Comment:
Before anything else, I just wanted to point out I am generally not happy with this kind of error messages as a user, because it's barely actionable. And I know existing test cases have this as well because it's from the json decoder. And unless we reinvent that decoder somewhat, we will have to live with this.

Anyway, I think there is no action item here. Just wanted to complain about it xD

Reply:
The errors are a at least a little more descriptive, the test is just doing prefix checking on the messages